src/components/Editor/Sidebar/LineButton.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useCallback, useState, MouseEvent } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+
+import Badge from '@material-ui/core/Badge';
+import Avatar from '@material-ui/core/Avatar';
+import IconButton from '@material-ui/core/IconButton';
+import BrushIcon from '@material-ui/icons/Brush';
+import Tooltip from '@material-ui/core/Tooltip';
+import Box from '@material-ui/core/Box';
+import { createStyles, makeStyles } from '@material-ui/core/styles';
+
+import Popover from 'components/commons/Popover';
+import { AppState } from 'app/rootReducer';
+import { Tool, setTool, Color, setColor } from 'features/boardSlices';
+
+const useStyles = makeStyles((theme) =>
+  createStyles({
+    button: {
+      marginLeft: 5,
+      padding: 7,
+    },
+    select: {
+      marginLeft: 5,
+      padding: 7,
+      color: theme.palette.primary.main,
+    },
+    box: {
+      width: '184px',
+      display: 'flex',
+      flexWrap: 'wrap',
+      '& > *': {
+        margin: theme.spacing(1),
+      },
+    },
+    colorIcon: {
+      '&:hover': {
+        cursor: 'pointer',
+      },
+      width: theme.spacing(3.1),
+      height: theme.spacing(3.1),
+    },
+    badge: {
+      backgroundColor: ({ color }: { color: Color }) => color,
+    },
+  }),
+);
+
+export default function LineButton() {
+  const dispatch = useDispatch();
+  const tool = useSelector((state: AppState) => state.boardState.tool);
+  const color = useSelector((state: AppState) => state.boardState.color);
+  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | undefined>();
+  const classes = useStyles({ color });
+
+  const handleOpen = useCallback(
+    (event: MouseEvent<HTMLButtonElement>) => {
+      if (tool !== Tool.Line) {
+        dispatch(setTool(Tool.Line));
+        dispatch(setColor(Color.Black));
+      }
+
+      setAnchorEl(event.currentTarget);
+    },
+    [tool],
+  );
+
+  const handleClose = useCallback(() => {
+    setAnchorEl(undefined);
+  }, []);
+
+  const handleSelectTool = (nextColor: Color) => () => {
+    dispatch(setTool(Tool.Line));
+    dispatch(setColor(nextColor));
+    handleClose();
+  };
```

Comment:
Is there any reason you didn't use `useCallback` only here?

Reply:
@hackerwins Thank you for your comment!
It is creating a function using currying.
This situation does not improve due to `useCallback` because the function is dynamically created when rendered.

If there is a performance problem, I will split the component and do memoization.